repository/src/main/java/org/alfresco/messaging/camel/configuration/ConnectionFactoryConfiguration.java

```diff
@@ -1,96 +0,0 @@
-/*
- * #%L
- * Alfresco Repository
- * %%
- * Copyright (C) 2005 - 2018 Alfresco Software Limited
- * %%
- * This file is part of the Alfresco software.
- * If the software was purchased under a paid Alfresco license, the terms of
- * the paid license agreement will prevail.  Otherwise, the software is
- * provided under the following open source license terms:
- *
- * Alfresco is free software: you can redistribute it and/or modify
- * it under the terms of the GNU Lesser General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * Alfresco is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
- * #L%
- */
-
-package org.alfresco.messaging.camel.configuration;
-
-import java.security.SecureRandom;
-
-import javax.jms.ConnectionFactory;
-
-import org.alfresco.encryption.AlfrescoKeyStore;
-import org.apache.activemq.ActiveMQConnectionFactory;
-import org.apache.activemq.ActiveMQSslConnectionFactory;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-
-/**
- * Configures the ActiveMQ connection factory for use.
- *
- * @author Gethin James
- */
-
-@Configuration
-public class ConnectionFactoryConfiguration
```

Comment:
This config class is necessary for other components, like event2 or even sync service. So why we removing this?

Reply:
@jamalkm please see my comment: https://github.com/Alfresco/alfresco-community-repo/pull/268#issuecomment-771534030

Reply:
@killerboot sorry, I don't get the comment!
That config class is only creating a connection using `ActiveMQConnectionFactory` for best performance rather the generic JMS connection factory or others. It point to localhost because this is how the property is set:
`messaging.broker.url=failover:(tcp://localhost:61616)?timeout=3000&jms.useCompression=true`
So in production we override this. For example, when using Amazon ActiveMQ.

This connection factory then plugged in into an ActiveMQ `PooledConnectionFactory`, then we use the `PooledConnectionFactory` to create Camel component like `amqp`. So repository `event2` is using the `amqp` component to configure Camel route.

Reply:
I don't disagree with any of that, but this is how it worked with previous camel. What happens now is that there is a separate component class ActiveMQComponent that controls the pool and connection factory. When dsync starts to use the events, those are going to localhost regardless of what is set into messaging.broker.url as the ActiveMQComponent is not using any of ConnectionFactoryConfiguration by default. The same applies to messaging.broker.connections.max=8, The thread dump shows 360+ threads for activemq, so the pool is not configured correctly.

You can observe the behaviour if you build the latest repository and then try to run SyncServiceAPITests in acs-packaging.
My changes are just the illustration of that. I'm not claiming that this class should be removed.

As a side note there is a complete copy of the configuration that creates another connection factory for desktop sync. Do we need 2 of those that are using the same ActiveMQ? https://github.com/Alfresco/dsync-services/blob/master/alfresco-device-sync-repo/src/main/java/org/alfresco/messaging/camel/configuration/DeviceSyncConnectionFactoryConfiguration.java

Reply:
I see, thanks for the clarification.
I think we need to simplify the ActiveMQ connection and use that connection throughout the Repository and any AMP that requires it.  
To answer your question re: 
> Do we need 2 of those that are using the same ActiveMQ? 

I think it's not required in the Desktop Sync AMP as long as they use the one from the repository.

Reply:
Also, I'll do a test without that class and I'll let you know the result.

Reply:
The redundant JMS configuration can be removed safely, this is a green build test https://travis-ci.com/github/Alfresco/dsync-services/builds/216040343